frontend/src/app/main/ui/workspace/viewport/outline.cljs

```diff
@@ -92,18 +92,17 @@
         edition     (obj/get props "edition")
         zoom        (obj/get props "zoom")
 
-        outlines    (set/union selected hover)
-
-        shapes      (d/concat-set
-                     (->> outlines
-                          (filter #(not= edition %))
-                          (map #(get objects %))
-                          (filterv show-outline?)
-                          (filter some?))
-                     ;; outline highlighted shapes even if they are hidden or blocked
-                     (->> highlighted
-                          (filter #(not= edition %))
-                          (map #(get objects %))
-                          (filter some?)))]
+        lookup      (d/getf objects)
+        edition?    (fn [o] (= edition o))
+
+        shapes      (-> #{}
+                        (into (comp (remove edition?)
+                                    (keep lookup)
+                                    (filter show-outline?))
+                              (set/union selected hover))
+                        (into (comp (remove edition?)
+                                    (keep lookup))
+                              highlighted))]
```

Comment:
Allbeit a more declarative approach than d/concat-set, surprisingly looks pretty readable. Perhaps due to raising a level of abstraction by giving anonymous functions names

Reply:
yep, this is probably the main reason, sometimes giving a good alias for a posible annonymous function makes thins more easy readable on first & quick look